service/src/main/kotlin/fi/espoo/evaka/reports/AttendanceReservationReportController.kt

```diff
@@ -79,7 +79,10 @@ private fun Database.Read.getAttendanceReservationReport(
           round(sum(r.service_need_factor * r.assistance_need_factor) / 7, 1) AS staff_count_required
         FROM generate_series(:start, :end + interval '1 day' - interval '1 second', interval '30 minutes') dateTime
         CROSS JOIN daycare u
-        LEFT JOIN reservations r ON r.date + date_trunc('hour', r.start_time) + date_part('minute', r.start_time)::int / 30 * interval '30 minutes' = dateTime
+        LEFT JOIN reservations r ON dateTime BETWEEN
+          r.date + date_trunc('hour', r.start_time) + date_part('minute', r.start_time)::int / 30 * interval '30 minutes'
+          AND
+          r.date + date_trunc('hour', r.end_time + interval '30 minutes' - interval '1 second') + date_part('minute', r.end_time + interval '30 minutes' - interval '1 second')::int / 30 * interval '30 minutes'
```

Comment:
I think this would be easier to understand in this form
```
LEFT JOIN reservations r ON dateTime BETWEEN
  r.date + r.start_time - interval '30 minutes' + interval '1 second'
  AND
  r.date + r.end_time + interval '30 minutes' - interval '1 second'
```

Reply:
Good find, fixed!